docs/proposals/DNSPolicy.md

```diff
@@ -0,0 +1,280 @@
+
+
+## None Goals
+
+- Allow different DNS policies for different listeners. Although this may be something we look to support in the future, currently policy attachment does not allow for this type of targeting. This means a DNSPolicy is applied for the whole gateway currently. 
+- Define how health checks should work, this will be part of a separate proposal
+
+
+## Terms
+
+- **managed listener**: This is a listener with a host backed by a DNS zone managed by the multi-cluster gateway controller
+- **hub cluster**: control plane cluster that managed 1 or more spokes
+- **spoke cluster**: a cluster managed by the hub control plane cluster. This is where gateway are instantiated
+
+## Proposal
+
+Provide a control plane DNSPolicy API that uses the idea of direct [policy attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment) from gateway API that allows a load balancing strategy to be applied to the DNS records structure for any managed listeners being served by the data plane instances of this gateway. 
+The DNSPolicy also covers health checks that inform the DNS response but that is not covered in this document.
+
+Below is a draft API for what we anticipate the DNSPolicy to look like
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+spec:
+  targetRef: # defaults to gateway gvk and currrent namespace
+    name: gateway-name
+  health:
+   ...
+  loadBalancing:
+    weighted: # always requird
+     default: 10  #always required
+     custom: #optional
+     - value: AWS  #optional with both GEO and weighted. With GEO the custom weight is applied to gateways within a Geographic region
+       weight: 10
+     - value: GCP
+       weight: 20
+    GEO: #optional
+      default: IE # required with GEO. Choses a default DNS response when no particular response is defined for a request from an unknown GEO.
+```  
+
+### Available Load Balancing Strategies  
+
+GEO and Weighted load balancing are well understood strategies and this API effectively allow a complex requirement to be expressed relatively simply and executed by the gateway controller in the chosen DNS provider. Our default policy will execute a "Round Robin" weighted strategy which reflects the current default behaviour.
+
+With the above API we can provide weighted and GEO and weighted within a GEO. A weighted strategy with a minimum of a default weight is always required and the simplest type of policy. The multi-cluster gateway controller will set up a default policy when a gateway is discovered (shown below). This policy can be replaced or modified by the user.  A weighted strategy can be complimented with a GEO strategy IE they can be used together in order to provide a GEO and weighted (within a GEO) load balancing. By defining a GEO section, you are indicating that you want to use a GEO based strategy (how this works is covered below).
```

Comment:
```
A weighted strategy with a minimum of a default weight is always required and the simplest type of policy
```
Out of curiousity, are we setting a weight value on endpoints in gateways that don't have an attached policy? If so, what value do we use?

Reply:
We create a default policy now, so there will always be a policy

Reply:
So the default policy is attached to any gateway that has no attached custom policies?

Reply:
yes